modules/ding_mobilesearch/includes/admin.inc

```diff
@@ -0,0 +1,264 @@
+<?php
+
+/**
+ * @file
+ * Mobilesearch administration and plugins settings UI.
+ */
+
+/**
+ * Content exporting settings form.
+ */
+function ding_mobilesearch_settings($form, &$form_state) {
+  $form['export'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Export'),
+    '#description' => t('Specify the node types and menus which should be exported.'),
+  );
+
+  // Content types exporting settings.
+  $form['export']['types'] = array(
+    '#theme' => 'ding_mobilesearch_types_table',
+    '#tree' => TRUE,
+  );
+  $names = node_type_get_names();
+
+  $base_trigger = array(
+    'hook_node_insert' => t('Node hooks (insert, update, delete)'),
+  );
+  $module_enabled = system_list('module_enabled');
+  $workflow_enabled = !empty($module_enabled['workflownode']);
+  if ($workflow_enabled) {
+    $type_map = workflow_get_workflow_type_map();
+  }
+
+  $plugin_info = module_invoke_all('mobilesearch_plugin_info');
+  $plugins = array();
+  foreach ($plugin_info as $key => $info) {
+    $plugins[$key] = $info['title'];
+  }
+  foreach ($names as $machine_name => $name) {
+    $trigger = $base_trigger;
+    if ($workflow_enabled) {
+      if (!empty($type_map[$machine_name])) {
+        $wid = $type_map[$machine_name];
+        $workflow = workflow_load($wid);
+        foreach ($workflow->states as $state) {
+          if ($state->name == '(creation)') {
+            continue;
+          }
+          $trigger['hook_workflow__' . $state->name] = $state->state;
+        }
+      }
+    }
+
+    $key = 'mobilesearch_type__' . $machine_name;
+    $form['export']['types']['rows'][$machine_name]['type'] = array(
+      '#type' => 'checkbox',
+      '#title' => $name,
+      '#default_value' => variable_get($key, FALSE),
+    );
+    $form['export']['types']['rows'][$machine_name]['trigger'] = array(
+      '#type' => 'radios',
+      '#options' => $trigger,
+      '#default_value' => variable_get($key . '__trigger', 'hook_node_insert'),
+    );
+    $form['export']['types']['rows'][$machine_name]['plugin'] = array(
+      '#type' => 'select',
+      '#options' => $plugins,
+      '#default_value' => variable_get($key . '__plugin', NULL),
+    );
+  }
+
+  // Menus exporting settings.
+  $form['export']['menus'] = array(
+    '#theme' => 'ding_mobilesearch_menus_table',
+    '#tree' => TRUE,
+  );
+  $result = db_select('menu_custom', 'm')
+    ->fields('m', array('menu_name', 'title'))
+    ->orderBy('m.title')
+    ->execute();
+  $menus = array();
+  foreach ($result as $row) {
+    if (preg_match('/^menu-og/', $row->menu_name)) {
```

Comment:
As I understand the code now, this would not allow the libraries to export there menus? That may be the wanted effect as I don't know what the exported data is used for. So a comment about why libray menu and "group" menu are not exporteable would be nice?